packages/next/pages/_document.tsx

```diff
@@ -126,10 +151,9 @@ export class Head extends Component<
 
   context!: React.ContextType<typeof DocumentComponentContext>
 
-  getCssLinks(): JSX.Element[] | null {
-    const { assetPrefix, files, devOnlyCacheBusterQueryString } = this.context
-    const cssFiles =
-      files && files.length ? files.filter((f) => f.endsWith('.css')) : []
+  getCssLinks(files: DocumentFiles): JSX.Element[] | null {
```

Comment:
This makes it harder for users to extend `getCssLinks` function. When they want to insert custom css links and control the order of the resources.

Same for the `getPreloadMainLinks` too

Reply:
This shouldn't be touched by users.